### PR TITLE
Add setup script for WSL

### PR DESCRIPTION
**Pull request by gilbertsoft**

### CODE REVIEW

File: scripts/setup-wsl.ps1 [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        try {
+            choco
+        } catch {
+            # Chocolatey was not found, install it now
+            #Invoke-Expression ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+            Invoke-Expression "& { $(Invoke-RestMethod 'https://chocolatey.org/install.ps1') }"
+        }
+
+        # Install mkcert
+        choco install mkcert -y
```
Comment (gilbertsoft) [inline scripts/setup-wsl.ps1:133]: Not tested so far

File: scripts/setup-wsl.ps1 [hunk 2]
```diff
@@ -0,0 +1,153 @@
+# Copyright (c) Gilbertsoft LLC. All rights reserved.
```
Comment (rfay) [inline scripts/setup-wsl.ps1:1]: Please don't use a different license, thanks. Repo is Apache, https://github.com/drud/ddev/blob/master/LICENSE

File: scripts/setup-wsl.ps1 [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor 3072 # = [Net.SecurityProtocolType]::Tls12
+
+try {
+    # Check if mkcert is installed
+    try {
+        mkcert -version
+    } catch [System.Management.Automation.CommandNotFoundException] {
+        # Check if Chocolatey is installed
+        try {
+            choco
```
Comment (rfay) [inline scripts/setup-wsl.ps1:125]: This could also install docker if it's not there yet?

**Comment by rfay** [comment]
This can't work with the default ExecutionPolicy.

I used `Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Scope LocalMachine` to be able to run it.

**Comment by rfay** [comment]
On a brand new Win10 system this fails with 

```
C:\Users\rfay\tmp\setup-wsl.ps1 : The term 'wsl' is not recognized as the name of a cmdlet, function, script file, or
operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try
again.
At line:1 char:1
+ ./setup-wsl.ps1
+ ~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (wsl:String) [setup-wsl.ps1], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException,setup-wsl.ps1
```

File: scripts/setup-wsl.ps1 (hunk 2 above)
Comment (gilbertsoft) [reply scripts/setup-wsl.ps1:1]: Sorry, just copied and pasted ;)

File: scripts/setup-wsl.ps1 (hunk 3 above)
Comment (gilbertsoft) [reply scripts/setup-wsl.ps1:125]: Ofc yes. That's why I was asking what exactly I should plan to include.

**Comment by gilbertsoft** [comment]
@rfay Have a look at the examples at https://github.com/drud/ddev/pull/2390/files#diff-02f0f6676455a2dabdee39fa065b3f54R10 and https://github.com/drud/ddev/pull/2390/files#diff-02f0f6676455a2dabdee39fa065b3f54R13, this should work...

File: scripts/setup-wsl.ps1 [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+        # Install mkcert
+        choco install mkcert -y
+    }
+
+    # Setup mkcert
+    mkcert -install
+
+    # Set default WSL version
+    wsl --set-default-version 2
```
Comment (gilbertsoft) [inline scripts/setup-wsl.ps1:140]: Throws an exception if WSL isn't installed. So we have to catch it and show a meaningful warning to the user.

**Comment by rfay** [comment]
I'm going to close this for now, but really happy to open it up again, just a comment will do it. But it's quite stale and I know you have other things going. I sure welcome you back to it when you get time.

**Pull request closed without merging**